File: src/spirite/base/pen/behaviors/PixelBehavior.java
package spirite.base.pen.behaviors;

import spirite.base.brains.ToolsetManager.Tool;
import spirite.base.brains.ToolsetManager.ToolSettings;
import spirite.base.pen.Penner;
import spirite.base.pen.StrokeEngine;
import spirite.base.pen.StrokeEngine.Method;
import spirite.base.pen.StrokeEngine.StrokeParams;
import spirite.base.pen.StrokeEngine.StrokeParams.InterpolationMethod;

public class PixelBehavior extends StrokeBehavior {
	final int color;
	public PixelBehavior( Penner penner, int i) {
		super(penner);
		this.color = i;
	}
	@Override
	public void start() {
		ToolSettings settings = this.penner.toolsetManager.getToolSettings(Tool.PIXEL);
		StrokeEngine.StrokeParams stroke = new StrokeEngine.StrokeParams();
		stroke.setMethod( StrokeEngine.Method.PIXEL);
		stroke.setAlpha((float)settings.getValue("alpha"));
		stroke.setHard(true);
		stroke.setColor( color);
		stroke.setInterpolationMethod(InterpolationMethod.NONE);
		startStroke( stroke);
	}
}

File: src/main/kotlin/spirite/base/pen/behaviors/PixelBehavior.kt
package spirite.base.pen.behaviors

import rb.glow.Color
import spirite.base.brains.toolset.IToolsetManager
import spirite.base.graphics.drawer.IImageDrawer.IStrokeModule
import spirite.base.pen.Penner
import spirite.base.pen.stroke.StrokeParams
import spirite.base.pen.stroke.StrokeParams.InterpolationMethod.NONE
import spirite.base.pen.stroke.StrokeParams.Method.PIXEL


object PixelBehavior {
    class Stroke(penner: Penner, drawer: IStrokeModule, val color: Color) : StrokeBehavior(penner,drawer) {
        override fun makeStroke() = PixelBehavior.makeStroke(penner.toolsetManager, color)
    }


    private fun makeStroke(toolsetManager: IToolsetManager, color: Color) : StrokeParams {
        val settings = toolsetManager.toolset.Pen

        return StrokeParams(
                color,
                PIXEL,
                settings.mode,
                1f,
                settings.alpha,
                interpolationMethod = NONE)
    }
}